### PR TITLE
created skeleton for server

### PR DESCRIPTION
**Pull request by VamshiTeja**
Created basic server skeleton for our project. Please review and merge.

### CODE REVIEW

**Comment by sipian** [comment]
keep `merauctions.go` in a folder as `cmd/auctions/main.go`

File: server/routes.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package server
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func setupRoutes(router *gin.Engine) {
+	// It is good practice to version your API from the start
```
Comment (sipian) [inline server/routes.go]: remove comment

File: server/routes.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	v1 := router.Group("/api/v0.1")
```
Comment (sipian) [inline server/routes.go]: we are not having any versioning of APIs

Comment (VamshiTeja) [reply server/routes.go]: In future we might change these right, not right now but still i think it is still good to have

**Comment by sipian** [comment]
@VamshiTeja Use more readable branch names. This will help in knowing later what each branch does.

File: server/routes.go (hunk 1 above)
Comment (VamshiTeja) [reply server/routes.go]: sure

**Pull request merged**